GQ_Characterize.hpp

```diff
@@ -0,0 +1,75 @@
+#ifndef GQ_CHARACTERIZE_H
+#define GQ_CHARACTERIZE_H
+
+#include <eigen3/Eigen/Eigen>
+#include <cfloat>
+#include <iostream>
+#include <string>
+
+#include "geometry.hpp"
+
+enum GQ_TYPE {UNKNOWN = 0,
+              ELLIPSOID,
+              ONE_SHEET_HYPERBOLOID,
+              TWO_SHEET_HYPERBOLOID,
+              ELLIPTIC_CONE,
+              ELLIPTIC_PARABOLOID,
+              HYPERBOLIC_PARABOLOID,
+              ELLIPTIC_CYL,
+              HYPERBOLIC_CYL,
+              PARABOLIC_CYL};
+
+const std::vector<std::string> _gq_names = {"UNKNOWN",
+                                            "ELLIPSOID",
+                                            "ONE_SHEET_HYPERBOLOID",
+                                            "TWO_SHEET_HYPERBOLOID",
+                                            "ELLIPTIC_CONE",
+                                            "ELLIPTIC_PARABOLOID",
+                                            "HYPERBOLIC_PARABOLOID",
+                                            "ELLIPTIC_CYL",
+                                            "HYPERBOLIC_CYL",
+                                            "PARABOLIC_CYL"};
+
+extern const std::vector<std::string> _gq_names;
+
+class GQ_Characterize{
+
+protected:
+  // coefficients of the GQ
+  double A_,B_,C_,D_,E_,F_,G_,H_,J_,K_;
+  // the cannonical GQ type
+  GQ_TYPE type;
+  // translation from the canoncial GQ to final GQ
+  Vector3d translation;
+  // rotation matrix from canonical GQ to final GQ
+  double rotation_mat[9];
+  // gq transform
+  Transform transform_;
+
+  // tolerance used to determine
+  // if matrix determinant should be considered zero
+  const double gq_tol = 1e-6;
+  const double equivalence_tol = 1e-8;
+
+
+public:
+  // Constructor
```

Comment:
We'd better add a comment here for clarity, adding the names of the coefficients of the constructor.

Reply:
Good call, I hadn't noticed that this constructor's signature didn't have variable names!